### PR TITLE
Make the LLaVA / LLaVA-Next test guard explicit

### PR DESCRIPTION
**Pull request by qgallouedec**
The VLM trainer tests assert every trainable parameter changes after `trainer.train()`. For LLaVA / LLaVA-Next some params are skipped with a vague:

> `# For some reason, these params are not updated. … We should investigate this further, but for now we just skip these params.`

We investigated. Those params are not bugs: they are unreachable from the loss **by design**.

This PR turns the skip into a positive assertion that they stay frozen, and cleans up two spurious keys in the tiny-model generation scripts.

## Why those params are frozen

LLaVA stitches three off-the-shelf pieces together:

1. **Vision tower**: a frozen, pretrained CLIP vision transformer (24 encoder layers in real LLaVA, 2 in our tiny version).
2. **Multi-modal projector**: two small `Linear` layers that map vision features into the LM's embedding dim.
3. **Language model**: a pretrained LLM (Vicuna for LLaVA-1.5, Mistral for LLaVA-Next).

At forward time: image → CLIP → pick **one intermediate encoder layer's** hidden states (`vision_feature_layer=-2`, the second-to-last) → drop the CLS token → projector → splice the result into the LM's input embeddings at the `<image>` placeholder.

Taking features from an intermediate layer rather than the final one is the reason the last encoder layer (and `post_layernorm`) never reach the loss.

```mermaid
flowchart LR
    img[Image] --> pe[patch_embed]
    pe --> l0[layers.0]
    l0 --> l1[layers.1]
    l1 --> dots["…"]
    dots --> l22[layers.22]
    l22 --> l23[layers.23]
    l23 --> pln[post_layernorm]
    pln -.->|"pooler_output<br/>(unused)"| X([ ])

    l22 ==>|"hidden_states[-2]"| drop["drop CLS token"]
    drop ==> proj[multi_modal_projector]
    proj ==> lm

    text[Text tokens] --> lm[Language Model]
    lm --> out[Output]

    subgraph vt[Vision tower — CLIP-ViT-Large, 24 layers]
        pe
        l0
        l1
        dots
        l22
        l23
        pln
    end

    style l23 stroke:#c33,stroke-dasharray:4 3
    style pln stroke:#c33,stroke-dasharray:4 3
    style X stroke:#c33,stroke-dasharray:4 3
```

**Bold arrows** = the actual path features take to the LLM.
**Red dashed** = computed every forward, but the result never reaches the loss (`vision_feature_layer = -2`, pooler unused).

---

> [!NOTE]
> **Low Risk**
> Low risk: changes are limited to test expectations and tiny-model generation configs, with no production training logic modifications; primary risk is making CI stricter around LLaVA parameter-freezing behavior.
> 
> **Overview**
> Makes the LLaVA/LLaVA-Next VLM trainer tests *explicitly* validate expected frozen vision-tower parameters: instead of skipping certain vision params, `DPO`/`SFT`/`GRPO`/`RLOO` tests now assert that `encoder.layers.1` and `post_layernorm` **do not change** while all other parameters must update after `trainer.train()`.
> 
> Cleans up the LLaVA tiny-model generation scripts by removing unused/incorrect `vision_config` keys (`num_key_value_heads`, `embed_dim`) for both `llava` and `llava_next` generators.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 164b4718aef330aabe99f5be384a81634ab159cd. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: scripts/generate_tiny_models/for_conditional_generation/llava_for_conditional_generation.py [hunk 1]
```diff
@@ -36,8 +36,6 @@
     "num_hidden_layers": 2,
     "hidden_size": 16,
     "num_attention_heads": 4,
-    "num_key_value_heads": 2,
-    "embed_dim": 64,
```
Comment (qgallouedec) [inline scripts/generate_tiny_models/for_conditional_generation/llava_for_conditional_generation.py:40]: these keys aren't used, we can safely drop them

File: tests/test_dpo_trainer.py [hunk 2]
```diff
@@ -975,23 +975,24 @@ def test_train_vlm(self, model_id):
...
             new_param = trainer.model.get_parameter(n)
+            # LLaVA & LLaVA-Next: vision_feature_layer=-2 leaves the last encoder layer (layers.1 and post_layernorm
+            # without gradient by design. Assert they stay frozen — if they ever start training, the feature-selection
+            # plumbing has likely regressed.
+            if model_id in (
+                "trl-internal-testing/tiny-LlavaForConditionalGeneration",
+                "trl-internal-testing/tiny-LlavaNextForConditionalGeneration",
+            ) and ("encoder.layers.1" in n or "post_layernorm" in n):
+                assert torch.equal(param, new_param), f"Param {n} expected frozen by LLaVA design, but changed"
+                continue
```
Comment (qgallouedec) [inline tests/test_dpo_trainer.py]:
```suggestion
```

File: tests/test_sft_trainer.py [hunk 3]
```diff
@@ -571,23 +571,24 @@ def test_train_chunked_nll_loss_vlm(self, model_id):
...
             new_param = trainer.model.get_parameter(n)
+            # LLaVA & LLaVA-Next: vision_feature_layer=-2 leaves the last encoder layer (layers.1 and post_layernorm
+            # without gradient by design. Assert they stay frozen — if they ever start training, the feature-selection
+            # plumbing has likely regressed.
+            if model_id in (
+                "trl-internal-testing/tiny-LlavaForConditionalGeneration",
+                "trl-internal-testing/tiny-LlavaNextForConditionalGeneration",
+            ) and ("encoder.layers.1" in n or "post_layernorm" in n):
+                assert torch.equal(param, new_param), f"Param {n} expected frozen by LLaVA design, but changed"
+                continue
```
Comment (qgallouedec) [inline tests/test_sft_trainer.py]:
```suggestion
```

File: tests/test_sft_trainer.py [hunk 4]
```diff
@@ -1650,23 +1651,24 @@ def test_train_vlm(self, model_id):
...
```
Comment (qgallouedec) [inline tests/test_sft_trainer.py]:
```suggestion
```

**Review by albertvillanova — APPROVED** [review]
Thanks.

Please do not forget to merge the corresponding tiny model PRs:
- [x] https://huggingface.co/trl-internal-testing/tiny-LlavaForConditionalGeneration/discussions/4
- [x] https://huggingface.co/trl-internal-testing/tiny-LlavaNextForConditionalGeneration/discussions/7

File: tests/conftest.py [hunk 5]
```diff
@@ -69,6 +69,8 @@ def pytest_runtest_makereport(item, call):
 
 MODEL_REVISIONS = {
     # Add model_id: revision mappings here to test PRs
+    "trl-internal-testing/tiny-LlavaForConditionalGeneration": "refs/pr/4",
+    "trl-internal-testing/tiny-LlavaNextForConditionalGeneration": "refs/pr/7",
```
Comment (qgallouedec) [inline tests/conftest.py]:
```suggestion
```
